### PR TITLE
chore(Box): deprecate enums

### PR DESCRIPTION
**Pull request by rivka-ungar**
https://monday.monday.com/boards/3532714909/views/80492480/pulses/7225220262

### CODE REVIEW

File: packages/core/src/components/Box/Box.types.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export type Disabled = "opacityDisabled";
+
+export type Border = "border";
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:3]: Wasn't it removed?

File: packages/core/src/components/Box/Box.types.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export type Disabled = "opacityDisabled";
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:1]: What is this for? Can we avoid it? It doesn't seem to be used

File: packages/core/src/components/Box/Box.types.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export type Border = "border";
+
+export type BorderColor = "borderColorUiBorderColor" | "borderColorLayoutBorderColor";
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:5]: Any chance we can get rid of the `border` prefix here? Not a priory but if it's small maybe it's worth it

File: packages/core/src/components/Box/Box.types.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+export type BackgroundColor =
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:7]: Same with `bg` prefix

File: packages/core/src/components/Box/Box.types.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  | "bgPrimaryBackgroundColor"
+  | "bgSecondaryBackgroundColor"
+  | "bgGreyBackgroundColor"
+  | "bgAllgreyBackgroundColor"
+  | "bgInvertedColorBackground";
+
+export type BoxTextColor = "textPrimaryTextColor" | "textTextColorOnInverted" | "textSecondaryTextColor";
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:14]: Same with `text`

File: packages/core/src/components/Box/Box.types.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+export type BoxPropSize = "xs" | "small" | "medium" | "large" | "xl" | "xxl" | "xxxl";
+
+export type BoxPropSizeWithAuto = "auto" | BoxPropSize;
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:18]:
```suggestion
type BaseBoxSize = "xs" | "small" | "medium" | "large" | "xl" | "xxl" | "xxxl";

export type BoxSize = "auto" | BaseBoxSize;
```

File: packages/core/src/components/Box/Box.types.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+export type Shadow = Extract<BoxPropSize, "xs" | "small" | "medium" | "large">;
+
+export type Rounded = "small" | "medium" | "big";
```
Comment (talkor) [inline packages/core/src/components/Box/Box.types.ts:22]:
```suggestion
export type RoundedSize = "small" | "medium" | "big";
```

**Review by talkor — COMMENTED** [review]
Amazing work!!
Had some questions/sugestion

File: packages/core/src/components/Box/Box.types.ts (hunk 1 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts]: Yep, forgot to remove it here

File: packages/core/src/components/Box/Box.types.ts (hunk 2 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts]: Yep, removed it

File: packages/core/src/components/Box/Box.types.ts (hunk 3 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts]: Yes!

File: packages/core/src/components/Box/Box.types.ts (hunk 4 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts:3]: removed

File: packages/core/src/components/Box/Box.types.ts (hunk 5 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts]: removed

File: packages/core/src/components/Box/Box.types.ts (hunk 6 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts]: The BaseBoxSize should not be exported?

Comment (talkor) [reply packages/core/src/components/Box/Box.types.ts]: Only if it's used for a certain prop

**Review by talkor — APPROVED** [review]
Excellent!

File: packages/core/src/components/Box/Box.types.ts (hunk 6 above)
Comment (rivka-ungar) [reply packages/core/src/components/Box/Box.types.ts]: Yes, it is used by a bunch of the props

**Pull request merged**